src/zinolib/event_types.py

```diff
@@ -0,0 +1,268 @@
+from datetime import datetime, timedelta, timezone
+from typing import Optional, ClassVar, List, TypeVar
+import logging
+
+from pydantic import IPvAnyAddress
+from pydantic import BaseModel, validator, Field, ValidationError
+
+from .compat import StrEnum
+
+
+def utcnow():
+    return datetime.now(timezone.utc)
+
+
+class AdmState(StrEnum):
+    """
+
+    Historic states:
+
+    * "active" was used twice in 1998
+    """
+    # ACTIVE = "active"
+
+    CLOSED = "closed"
+    CONFIRM_WAIT = "confirm-wait"
+    IGNORED = "ignored"
+    OPEN = "open"
+    WAITING = "waiting"
+    WORKING = "working"
+
+
+class FlapState(StrEnum):
+    FLAPPING = 'flapping'
+    STABLE = 'stable'
+
+
+class BFDState(StrEnum):
+    ADMIN_DOWN = "adminDown"
+    DOWN = "down"
+    INIT = "init"
+    UP = "up"
+
+
+class PortState(StrEnum):
+    """
+    Historic states:
+
+    * "testing" was used six times from 2000 to 2001
+    * "admin0" was used ten times from 2000 to 2002
+    * "flapping" was used 22 times on 2016-07-13T09:55 UTC
+    * "notPresent" was used 161 times during 2003
+    * "5" was used 389 times from 1998 to 2000
+    * "dormant" was used 813 times from 2000 to 2006
```

Comment:
I suspect the "flapping" enum value is no longer used/supported since it was only ever used once, during one minute, since 1998. There is a separate enum for showing flapping which I assume was added after that happened.

Does the existing server push these values on blindly?

Reply:
We'll have to ask HE about this.

Reply:
Go right ahead and ask HE. Since many of the "strange" values exist only in old data and for limited periods, I would mostly assume they are due to old bugs or at least old versions of the Zino codebase, and do not necessarily represent today's situation.

Most certainly, the `5` value you saw in the data is identical to `dormant`: Zino likely got this state value from some ports, and some hiccup in Scotty prevented the raw value from being translated to the MIB's name for that value, `dormant`.